packages/preact-iso/router.js

```diff
@@ -176,7 +178,8 @@ export function Router(props) {
 
 		// The route is loaded and rendered.
 		if (wasPush) scrollTo(0, 0);
-		if (props.onLoadEnd) props.onLoadEnd(url);
+		if (props.onLoadEnd && isLoading.current) props.onLoadEnd(url);
+		isLoading.current = false;
 	});
```

Comment:
The bigger issue is here as we're basically calling this `layout-effect` on _ever single render_ rather than when it's needed, as this effect is inherently built on refs it's hard to give a reliable deps array, a reliable way would be to use a `state` for the loading times

Reply:
Hmm - that behavior is intentional though - the effect callback runs after every render to check whether that render suspended or not.

Reply:
Yes but suspending should be part of the observed state else you're hoping that the render cycle of your top-level router ties in with your suspended boundary, this implies that your `useReducer` and `lazy.setState` happen correctly